### DISCUSSION TITLE
Split the package for native Dart support

### DISCUSSION THREAD

**Discussion by TesteurManiak** [Floating-Dartists/matomo-tracker#90] [Ideas]
This is something I've been thinking about after seeing this issue on the repo we've initially forked: https://github.com/poitch/dart-matomo/issues/41

The idea would be to move Flutter code and dependencies to a separate package (e.g.: `matomo_tracker_flutter`) and only keep pure Dart code in `matomo_tracker`. The biggest change would be to convert this repository into a monorepo (probably managed using [melos](https://github.com/invertase/melos)) and redirect the userbase to the new Flutter package to keep using it easily inside their Flutter project.

This would allow for a better separation of responsibilities and make the whole logic useable inside 3rd party Dart frameworks.

**Comment by EPNW-Eric** [comment]
I agree that a separation would be nice, but I think it would take some effort to find a good solution.

A concern that comes to my mind is that there is a dependence on `shared_preferences` which is a flutter package. The dart-only-version would either have to provide a solution that works for flutter and non-flutter, or find some kind of abstraction (which optimally would then be implemented by the flutter version automatically, probably on the base of `shared_preferences`).

**Comment by TesteurManiak** [comment]
We could migrate from `shared_preferences` to `hive` (or `isar` when web support will work) as it is independent from the Flutter SDK.

As we already have the `LocalStorage` abstraction it wouldn't be too hard. (But not a priority for v4.0.0 maybe for a future v5.0.0)
